internal/agent/restore.go

```
@@ -160,35 +133,75 @@ func RunRestore(ctx context.Context) error {
-// restore agent's etcdutl differ in major.minor. Empty/unparseable inputs skip
-// the check (best-effort) rather than block a restore.
-func checkRestoreVersionCompat(clusterVersion, etcdutlVersion string) error {
-	cm, um := majorMinor(clusterVersion), majorMinor(etcdutlVersion)
-	if cm == "" || um == "" {
-		return nil
+func copyExecutable(src, dst string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("open %s: %w", src, err)
+	}
+	defer in.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `in.Close` is not checked

(errcheck)